packages/ui/src/components/Topic/EditTopicProperties/EditTopicProperties.tsx

```diff
@@ -87,10 +81,13 @@ export const EditTopicProperties: FunctionComponent<
     const tranformedValueInBytes = retentionSizeTransformer(
       customRetentionSizeValue
     );
+    const updatedPartitions = Array(partitions)
+      .fill(null)
+      .map((_, index) => ({ partition: index }));
     const transformedTopic: Topic = {
       ...topic,
       name: topicName,
-      partitions: partitions,
+      partitions: updatedPartitions,
```

Comment:
This is the same as using the partitions array to show a number. Why should this component know and deal with the complexity of generating the Topic object? We should have a TopicForm type that contains the bits that the user can configure that gets sent on save, and then on the mutation side the right data can be picked so to make the mutation work.